servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyHttpServer.java

```diff
@@ -349,19 +344,10 @@ public void onComplete() {
                         }));
 
                 final HttpRequestMethod requestMethod = request.method();
-                final HttpKeepAlive keepAlive = HttpKeepAlive.responseKeepAlive(request);
-                Single<StreamingHttpResponse> respSingle;
-                try {
-                    respSingle = service.handle(this, request, streamingResponseFactory());
-                } catch (Throwable cause) {
-                    respSingle = failed(cause);
-                }
-                Publisher<Object> respPublisher = respSingle
-                        .onErrorReturn(cause -> newErrorResponse(cause, executionContext.executor(),
-                                request.version(), keepAlive))
+                // Don't expect any exceptions from service because it's already wrapped with
+                // ExceptionMapperServiceFilterFactory
+                Publisher<Object> respPublisher = service.handle(this, request, streamingResponseFactory())
```

Comment:
The filtering approach works nicely in 0.42 because offloading is already moved into a filter. Will have more fun during cherry-pick for 0.41. I will either try to use `offloadService` method that works as a filter there OR apply the observer filter after offloading, meaning it won't see `RejectedExecutionException` and numbers won't be accurate due to offliading. Will see how it goes.
@bondolo how do you think, will it be hard to cherry-pick https://github.com/apple/servicetalk/pull/1710 into 0.41?

Reply:
I have created a cherry-pick PR that is my first attempt at a minimized version of the #1710 changes. We can either trim it further or merge it as-is.

Reply:
Reviewing the cherry-pick after a weekend I think it is fine.